test/test_synthetic.py

```diff
@@ -0,0 +1,407 @@
+
+        wm_bounds = aoi.calc_buffer_ray(self.los.getSensorDirection(), lookDir=self.los.getLookDirection())
+        self.wmObj.set_latlon_bounds(wm_bounds)
+        wm_fname  = make_weather_model_filename(self.WM, self.dt, self.wmObj._ll_bounds)
+        self.path_wm_real = op.join(self.wd, 'weather_files_real', wm_fname)
+
+        grid_spacing = 0.5
+        self.cube_spacing = np.round(grid_spacing/1e-5).astype(np.float32)
+        self.hgt_lvls     = np.arange(-500, 1500, 500)
+
+
+    def setup_region(self):
+        if self.reg == 'LA':
+            self.SNWE  = 33, 34, -118.25, -117.25
+            self.dt    = datetime(2020, 1, 30, 13, 52, 45)
+            self.orbit = self.wd + \
+                '/S1B_OPER_AUX_POEORB_OPOD_20210317T025713_V20200129T225942_20200131T005942.EOF'
+
+        elif self.reg == 'Fort':
+            self.SNWE = -4.0, -3.5, -38.75, -38.25
+            self.dt   = datetime(2019, 11, 17, 20, 51, 58)
+            self.orbit = self.wd + \
+                '/S1A_OPER_AUX_POEORB_OPOD_20210315T014833_V20191116T225942_20191118T005942.EOF'
+
+
+    def make_config_dict(self):
+        dct = {
+            'aoi_group': {'bounding_box': list(self.SNWE)},
+            'height_group': {'height_levels': self.hgt_lvls.tolist()},
+            'time_group': {'time': self.ttime, 'interpolate_time': False},
+            'date_group': {'date_list': datetime.strftime(self.dt, '%Y%m%d')},
+            'cube_spacing_in_m': str(self.cube_spacing),
+            'los_group': {'ray_trace': True, 'orbit_file': self.orbit},
+            'weather_model': self.WM,
+            'runtime_group': {'output_directory': self.wd},
+        }
+        return dct
+
+
+def dl_real(reg='LA', mod='GMAO'):
+    """ Download the real weather model to overwrite
+
+    This 'golden dataset' shouldnt be changed
+    """
+    SAobj = StudyArea(reg, mod)
+    dct_cfg = SAobj.make_config_dict()
+    # set the real weather model path and download only
+    dct_cfg['runtime_group']['weather_model_directory'] = \
+            op.dirname(SAobj.path_wm_real)
+    dct_cfg['download_only'] = True
+
+    cfg = update_yaml(dct_cfg)
+
+    ## run raider to download the real weather model
+    cmd  = f'raider.py {cfg}'
+
+    proc = subprocess.run(cmd.split(), stdout=subprocess.PIPE, universal_newlines=True)
+    assert np.isclose(proc.returncode, 0)
+    return
+
+
+def test_hydrostatic_eq(reg='Fort', mod='GMAO'):
+    """ Test the hydrostatic eqn by setting t=p and comparing to ray
+
+    The ray length is computed here (length_of_ray; m), and in raider (m * K/Pa)
+    ray length computed here is then scaled by constant K/Pa
+    ray length from raider is unscaled from parts per million
+    We check they are both large enough for meaningful numerical comparison (>1)
+    We then compute residual and normalize it we theoretical ray length
+    We ensure that normalized residual is not significantly different from 0
+        significantly different = 7 decimal places
```

Comment:
Please mention something to the effect: The integration of the delay along a ray from the weather model node to the sensor is the length of the ray when the integrand is constant.

Reply:
I mean the description "Test the hydrostatic eqn by setting t=p and comparing to ray" should be expanded.

Reply:
phrase the hypothesis of what you are testing and then explain how you test it.

The hydrostatic delay reduces to an integral along the ray path when P=T. However the constants k1 and scaling of 10-6 woill remain present leading to a result of ray-lenght *k1*10^-6. 

Below we do the following steps: